deployments/spacelift/dpe-k8s/main.tf

```diff
@@ -201,3 +211,43 @@ resource "spacelift_aws_integration_attachment" "k8s-deployments-aws-integration
   read           = true
   write          = true
 }
+
+
+resource "spacelift_stack" "auth0" {
+  github_enterprise {
+    namespace = "Sage-Bionetworks-Workflows"
+    id        = "sage-bionetworks-workflows-gh"
+  }
+
+  depends_on = [
+    spacelift_space.dpe-space
+  ]
+
+  administrative          = false
+  autodeploy              = var.auto_deploy
+  branch                  = var.git_branch
+  description             = "Stack used to create and manage Auth0 for authentication"
+  name                    = var.auth0_stack_name
+  project_root            = var.auth0_stack_project_root
+  repository              = "eks-stack"
+  terraform_version       = var.opentofu_version
+  terraform_workflow_tool = "OPEN_TOFU"
+  space_id                = spacelift_space.dpe-space.id
+  additional_project_globs = [
+    "deployments/"
+  ]
+}
+
+resource "spacelift_stack_destructor" "auth0-stack-destructor" {
+  stack_id = spacelift_stack.auth0.id
+}
+
+
+resource "spacelift_environment_variable" "auth0-stack-environment-variables" {
+  for_each = local.auth0_stack_variables
+
+  stack_id   = spacelift_stack.auth0.id
+  name       = "TF_VAR_${each.key}"
+  value      = try(tostring(each.value), jsonencode(each.value))
+  write_only = false
+}
```

Comment:
This creates a new stack within Spacelift that is responsible for handling Auth0 stuff only; It lowers the concerns each stack is responsible for.